src/plugins/explore/public/components/query_panel/components/editor_stack/prompt_editor.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { EuiIcon } from '@elastic/eui';
+import { getEditorConfig, LanguageType } from './shared';
+import { ReusableEditor } from './resuable_editor';
+
+export interface PromptEditorProps {
+  languageType: LanguageType;
+  prompt: string;
+  isPromptReadOnly: boolean;
+  onChange: (value: string) => void;
+  onPromptRun: (queryString?: string) => void;
+  onPromptEdit: () => void;
+  onClearEditor: () => void;
+}
+
+export const PromptEditor: React.FC<PromptEditorProps> = ({
+  languageType,
+  prompt,
+  isPromptReadOnly,
+  onChange,
+  onPromptRun,
+  onPromptEdit,
+  onClearEditor,
+}) => {
+  const editorConfig = getEditorConfig(languageType);
+  const langText = languageType !== 'nl' ? 'query' : 'prompt';
```

Comment:
nitpick: prefer `languageText` over `langText`.

But besides that point, Is `languageText` the correct naming here? It seems like it should be called `queryType` or something. Also could we not hardcode `nl` `query` or `prompt`? These should be enum values